torch/testing/_internal/common_utils.py

```diff
@@ -801,204 +801,304 @@ def is_integral(self, dtype):
+            # Short-circuits on identity
+            if x == y or (equal_nan and x != x and y != y):
                 return
-            super().assertLessEqual(abs(x - y), atol, message)
+
+            # Acquires rtol and atol based on number type
+            if isinstance(x, complex) or isinstance(y, complex):
+                rtol = self.dtype_precisions[torch.complex64][0] if rtol is None else rtol
+                atol = self.dtype_precisions[torch.complex64][1] if atol is None else atol
+            elif isinstance(x, float) or isinstance(y, float):
+                rtol = self.dtype_precisions[torch.float32][0] if rtol is None else rtol
+                atol = self.dtype_precisions[torch.float32][1] if atol is None else atol
+            else:
+                rtol = 0 if rtol is None else rtol
+                atol = 0 if atol is None else atol
+
+            self.assertTrue(abs(x - y) <= (atol + rtol * abs(y)), msg=message)
         else:
-            super().assertEqual(x, y, message)
+            super().assertEqual(x, y, msg=message)
 
-    def assertAlmostEqual(self, x, y, places=None, msg='', delta=None, allow_inf=None):
+    def assertAlmostEqual(self, x, y, places=None, msg='', delta=None):
         prec = delta
         if places:
             prec = 10**(-places)
-        self.assertEqual(x, y, msg, atol=prec, allow_inf=allow_inf)
+        self.assertEqual(x, y, msg, atol=prec)
 
     def assertNotEqual(self, x, y, message='', *, atol=None):
-        if not isinstance(message, str):
-            raise Error("fix this test, message should be a string")
-        if atol is None:
-            (_, atol) = self.get_default_tolerance(x, y)
-
-        if isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor):
-            if x.size() != y.size():
-                super().assertNotEqual(x.size(), y.size())
-            self.assertGreater(x.numel(), 0)
-            y = y.type_as(x)
-            y = y.cuda(device=x.get_device()) if x.is_cuda else y.cpu()
-            nan_mask = x != x
-            if torch.equal(nan_mask, y != y):
-                diff = x - y
-                if diff.is_signed():
-                    diff = diff.abs()
-                diff[nan_mask] = 0
-                # Use `item()` to work around:
-                # https://github.com/pytorch/pytorch/issues/22301
-                max_err = diff.max().item()
-                self.assertGreater(max_err, atol, message)
-        elif type(x) == str and type(y) == str:
-            super().assertNotEqual(x, y)
-        elif is_iterable(x) and is_iterable(y):
-            super().assertNotEqual(x, y)
-        else:
-            try:
-                self.assertGreater(abs(x - y), atol, message)
-                return
-            except (TypeError, AssertionError):
-                pass
-            super().assertNotEqual(x, y, message)
+        threw = False
+        try:
+            self.assertEqual(x, y, message, atol=atol)
```

Comment:
similarly here I'd like to improve on our error messages, possibly in a follow-up PR, so that at least in a default case we're providing more information about specific mismatches in failed tests.

Reply:
I'm not sure what we could say in this case?

Reply:
I guess for the not-equal case it's nothing fancy, since it wouldn't be a specific index to call out, but I think we could display something nicer than "False != True." For the python built-in tests we see:
```
unittest.TestCase().assertNotEqual(1, 1)
AssertionError: 1 == 1
```

At some point I think we should add rtol here too.

nit: this code might be simpler with assertRaisesRegex.

Reply:
There are some cases, like comparing scalars, where we could be more informative, true. In general it's hard to imagine for tensors. 

You're right we can probably rewrite this to use assertRaises.

Reply:
Whenever I see an assertion fail with `True != False`, pointing to a line of code that says something like `self.assertNotEqual(x, y)`, I interpret the message as incorrectly stating that x and y are booleans, and that the booleans weirdly were actually unequal.  Even just a correct indication of type like "tensors unexpectedly equal" would seem nicer to me.

I'm fine with iterating on the notEquals message later though, the equals one seems more important where there's specific useful information we can include.

Reply:
I see what you're getting at. Yeah that should be a follow-up: it'll be a little tricky to get nice behavior there.

Reply:
(Funnily enough, that error message may be slightly improved since it should now say something like "AssertionError not raised" on the assertEqual line.)

Reply:
I probably should have put in a unit test validating we were getting the expected message when I added the assert_allclose usage the first time.

Reply:
@nairbv Take a look now, I added multiple debug messages (and tests that demonstrate the behavior).